WordPress/gutenberg
# Site Editor: Prevent classic theme from accessing global style menu

## What?

This PR fixes an issue where classic themes could unintentionally access global styles when they get access to the Patterns page in the site editor in the future.

![classic-theme-global-styles](https://github.com/WordPress/gutenberg/assets/54422211/1298ce4a-b350-45d4-bb15-20913739c1fe)

## Why?

As indicated in [this comment](https://github.com/WordPress/gutenberg/issues/52150#issuecomment-1743235386), the goal is for all Classic themes to have access to the Site Editor's Patterns page in the future.

As of WP6.5, access to this Pattern page is allowed internally, but I discovered that when I open the pattern editor canvas I can access the Global Styles menu.

My understanding is that the Global Styles is not expected to support the classic theme in the future, so this menu should not appear.

## How?

Support for global styles currently relies on `supportsTemplatePartsMode` being `false`. This variable is determined by whether the theme is not classic and supports template parts:

https://github.com/WordPress/wordpress-develop/blob/5c4b4887bf088b91df4f5e467b8982fa3b5441ec/src/wp-admin/site-editor.php#L64

The fact that `supportsTemplatePartsMode` is `false` means that the following conditions are met:

```
wp_is_block_theme() || !current_theme_supports( 'block-template-parts' )
```

If the classic theme does not have template parts opted in, this condition is `false`, which means `supportsGlobalStyles` is `true`.

My understanding is that whether it supports the Global Styles or not is completely consistent with whether it's a block theme, so I updated it to only check if it's a block theme instead of `supportsTemplatePartsMode`.

## Testing Instructions

- In WP Core, [the ticket](https://core.trac.wordpress.org/ticket/58827) allowing classic themes to access the Site Editor has not yet been committed. Comment out two places in `wp-admin/site-editor.php` to temporarily allow access:
 ```php
  if ( ! ( current_theme_supports( 'block-template-parts' ) || wp_is_block_theme() ) ) {
  	// wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
  }

  $is_template_part        = isset( $_GET['postType'] ) && 'wp_template_part' === sanitize_key( $_GET['postType'] );
  $is_template_part_path   = isset( $_GET['path'] ) && 'wp_template_partall' === sanitize_key( $_GET['path'] );
  $is_template_part_editor = $is_template_part || $is_template_part_path;

  if ( ! wp_is_block_theme() && ! $is_template_part_editor ) {
  	// wp_die( __( 'The theme you are currently using is not compatible with the Site Editor.' ) );
  }
```
- Enable classic theme (TT1).
- Access `http://localhost:8888/wp-admin/site-editor.php?path=%2Fpatterns` and create a pattern.
- The Global Styles icon should not appear in the sidebar.
